File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int candidate;
        int count=0;
        for(int num:nums){
            if(count == 0){
                candidate = num;
            }
            if(num == candidate){
                count++;
            }
            else count --;
        }
        return candidate;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cn = 0;
        int el = 0;
        for(int i = 0; i < nums.length; i++){
            if(cn == 0){
                el = nums[i];
            }
            if(nums[i] == el)
                cn++;
            else
                cn--;
        }
        return el;
    }
}